File: python/字符串/1614. 括号的最大嵌套深度.py
class Solution:
    def maxDepth(self, s: str) -> int:
        stack = []
        res = 0
        for i in s:
            if i == '(':
                stack.append(i)
            elif i == ')':
                stack.pop()
            res = max(res,len(stack))
        return res

File: go/string/1614. 括号的最大嵌套深度.go
package string

func maxDepth(s string) int {
	var stack []rune
	var maxDepth int
	for _, v := range s {
		if v == '(' {
			stack = append(stack, v)
		} else if v == ')' {
			stack = stack[:len(stack)-1]
		}
		if len(stack) > maxDepth {
			maxDepth = len(stack)
		}
	}
	return maxDepth
}
